internal/pkg/service/stream/definition/repository/source/source_version.go

```diff
@@ -54,6 +54,15 @@ func (r *Repository) RollbackVersion(k key.SourceKey, now time.Time, by definiti
 			// Prepare the new value
 			versionDescription := fmt.Sprintf(`Rollback to version "%d".`, targetVersion.Version.Number)
 			updated = deepcopy.Copy(*targetVersion).(definition.Source)
+
+			// Preserve Disabled and Deleted states from the latest version.
+			// Without this it would be possible to bypass the Enable/Undelete event and avoid billing.
+			updated.Disabled = deepcopy.Copy(latestVersion.Disabled).(*definition.Disabled)
+			updated.Enabled = deepcopy.Copy(latestVersion.Enabled).(*definition.Enabled)
+			updated.Deleted = deepcopy.Copy(latestVersion.Deleted).(*definition.Deleted)
+			updated.Undeleted = deepcopy.Copy(latestVersion.Undeleted).(*definition.Undeleted)
```

Comment:
What do you think about this? The intention is to make it so that the client can't avoid billing by disabling a source and then rollbacking to it's enabled state or vice versa being billed incorrectly after rollbacking from enabled to disabled state. We could alternatively try to detect these changes on rollback and fire the relevant events instead. Which way do you think is better?

Reply:
I don't mind this approach, just make sure the code is not duplicated that much. Create some "deep-partial-copy" function